### PR TITLE
Update of simple mask estimator example

### PR DESCRIPTION
**Pull request by jensheit**
Restructures the mask estimator example to fit the example script structure.

Adds evaluation script and README

Adds normalization and recurrent layer to mask estimator to improve enhancement results.

### CODE REVIEW

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+Simple Mask Estimator
+=============
+
+This directory contains scripts to train and evaluate a simple mask estimator
+inspired by .
+
+Results
+-------
+
+The simple mask estimator model can achieve the following results:
```
Comment (boeddeker) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:10]: Can you mention, that it is on the Chime3 6ch track?
Are the lines `observed` and `masked` on the 1ch track or on the first channel of the 6ch track?

Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:10]: yes, I'll add that, its on the first channel of the 6ch track

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Evaluation
+----------
+
+The evaluation requires `dlp_mpi` and `pb_bss` as additional dependencies.
+`dlp_mpi` can be installed via `pip install dlp_mpi` and `pb_bss` is available at [github.com/fgnt/pb_bss](github.com/fgnt/pb_bss).
+The evaluation can be started by
+
+```bash
+$ STORAGE_ROOT=/path/to/your/storage; python -m padertorch.contrib.examples.speech_enhancement.simple_mask_estimator.evaluate
```
Comment (boeddeker) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:37]: Since you mention `dlp_mpi`, you should give an example, for an MPI call, e.g. `mpirun -np $(nproc --all) python -m ...`

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+```
+It always evaluates the latest model in the specified STORAGE_ROOT
```
Comment (boeddeker) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:39]: Can you give an example, how to evaluate one specific model? I think the common use case is, that you want to evaluate one particular model.

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/evaluate.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        .map(change_example_structure)
+
+
+def evaluate():
+    model = SimpleMaskEstimator(513)
+    task_dir = STORAGE_ROOT / 'speech_enhancement'
+    dirs = list(task_dir.glob('simple_mask_estimator_*'))
+    latest_id = sorted([int(path.name.split('_')[-1]) for path in dirs])[-1]
+    model_dir = task_dir / f'simple_mask_estimator_{latest_id}'
+    checkpoint_path = model_dir / 'checkpoints' / 'ckpt_best_loss.pth'
```
Comment (boeddeker) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/evaluate.py:56]: Can you resolve the symlink and print something to stdout? e.g. `print(f'Start to evaluate the checkpoint {checkpoint_path} and will write the evaluation result to ...')`

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md (hunk 3 above)
Comment (boeddeker) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:39]: Or alternatively, can you mention here, that we dropped a CLI to keep this example as simple as possible?

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md (hunk 1 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md]: done

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md (hunk 2 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md]: done

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/evaluate.py (hunk 4 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/evaluate.py:56]: done

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+Simple Mask Estimator
+=============
+
+This directory contains scripts to train and evaluate a simple mask estimator
+inspired by .
```
Comment (thequilo) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:5]: One space too much

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+```bash
+$ STORAGE_ROOT=/path/to/your/storage; mpiexec -n $(nproc --all) python -m padertorch.contrib.examples.speech_enhancement.simple_mask_estimator.evaluate
+```
+It always evaluates the latest model in the specified STORAGE_ROOT
+
+References
+----------
+
+  [1] J. Heymann and L. Drude and A. Chinaev and R. Haeb-Umbach.
```
Comment (thequilo) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:53]: You never mention this reference in the above text

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md (hunk 3 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:42]: done

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py [hunk 7]
```diff
@@ -1,24 +1,23 @@
...
 from padercontrib.database.chime import Chime3
-from pb_bss.extraction.mask_module import biased_binary_mask
-import padertorch as pt
+from padercontrib.database.iterator import AudioReader
 from padertorch.summary import mask_to_image, stft_to_image
+from pb_bss.extraction.mask_module import biased_binary_mask
 
+ex = Experiment('Simple Mask Estimator')
 
 class SimpleMaskEstimator(pt.Model):
```
Comment (thequilo) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py]: Move this into a separate file `model.py` to be consistent with the other examples?

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py [hunk 8]
```diff
@@ -1,24 +1,23 @@
...
-import torch
-
 import paderbox as pb
 import padercontrib.database.keys as K
+import padertorch as pt
+import torch
+from sacred import Experiment
 from padercontrib.database import JsonAudioDatabase
-from padercontrib.database.iterator import AudioReader
 from padercontrib.database.chime import Chime3
```
Comment (thequilo) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py:14]: This only works in our environment. The other examples were modified to at least work without `padercontrib`, i.e., they take a JSON file as parameter and use `pb.io.load_audio` for audio loading.

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py (hunk 7 above)
Comment (thequilo) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py]: Okay, you already did this while I was reviewing

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md (hunk 6 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/README.md:53]: the space you commented about above should have been the reference. Its fixed now

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py (hunk 7 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py]: done

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/model.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+class SimpleMaskEstimator(pt.Model):
+    def __init__(self, num_features, num_units=1024, dropout=0.5,
+                 activation='elu'):
+        """
+
+        :param num_features: number of input features
+        :param num_units: number of units in linear layern
+        :param dropout: dropout forget ratio
+        :param activation:
```
Comment (thequilo) [inline padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/model.py:14]: Use Google-Style docstring? A great part of the code in padertorch uses google style, so I think that new code that we add should also use this style

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py (hunk 8 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/train.py]: done

File: padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/model.py (hunk 9 above)
Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/simple_mask_estimator/model.py]: done

File: padertorch/contrib/examples/speech_enhancement/mask_estimator/README.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+Training
+--------
+
+A storage root must be set with `export STORAGE_ROOT=/path/to/your/storage`.
+After installing `padertorch`, a training can for example be started with
+
+```bash
+$ STORAGE_ROOT=/path/to/your/storage; python -m padertorch.contrib.examples.speech_enhancement.simple_mask_estimator.train with database_json=/path/to/json
+```
```
Comment (michael-kuhlmann) [inline padertorch/contrib/examples/speech_enhancement/mask_estimator/README.md:30]: I guess the JSON needs a certain structure (e.g., certain keys) to work with the example. Is it possible to give a short description of the JSON's structure?

Comment (jensheit) [reply padertorch/contrib/examples/speech_enhancement/mask_estimator/README.md:30]: done

**Review by michael-kuhlmann — APPROVED** [review]

File: padertorch/modules/recurrent.py [hunk 11]
```diff
@@ -12,7 +12,8 @@ def __init__(
             num_layers: int = 1,
             bidirectional: bool = False,
             dropout: float = 0.,
-            batch_first: bool = True
+            batch_first: bool = True,
+            save_states: bool = False
```
Comment (boeddeker) [inline padertorch/modules/recurrent.py:16]: Should the default for the class `StatefulLSTM` be, that it is statefull?

Comment (jensheit) [reply padertorch/modules/recurrent.py:16]: you are right, I'll switch that and remove the defaults for input_size and hidden_size since they are quiet random

**Review by boeddeker — APPROVED** [review]
